services/QuillLMS/app/services/pdf_file_builder.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class PdfFileBuilder < ApplicationService
+  ENCODING = 'ascii-8bit'
+  TEMPFILE_NAME = 'temp.pdf'
+
+  attr_reader :data, :template
+
+  def initialize(data, template)
+    @template = template
+    @data = data
+  end
+
+  def run
+    tempfile.write(pdf)
+    tempfile.rewind
+    tempfile
```

Comment:
Not completely necessary, but when this is used, it would be good to call `unlink` on the `tempfile` to clean it up.

Reply:
My plan was to use `unlink` in a separate worker PR where this builder is actually used (or add a block that does the cleanup in here).  At this point I'm punting on this decision.